sonar-text-plugin/src/test/java/org/sonar/plugins/secrets/utils/SecretsRegexTest.java

```diff
@@ -0,0 +1,284 @@
+        "S6694, id: mongodb-cli-quoted-outer, location: post, the regex: `^<[a-z \\-]+>$`, violating rule: S5867: Replace this character range with a Unicode-aware character class.\n"
+        +
+        "S6694, id: mongodb-cli-quoted-outer, location: post, the regex: `^--[a-z\\-]+$`, violating rule: S5867: Replace this character range with a Unicode-aware character class.\n");
+  }
+
+  private static Set<String> listOfYamlFiles() {
+    Path specificationsDirectory = Path.of(CONFIGURATION_FILES_PATH);
+    String[] extensionsToSearchFor = new String[] {"yaml"};
+    Collection<File> files = FileUtils.listFiles(new File(specificationsDirectory.toUri()), extensionsToSearchFor, false);
+    return files.stream().map(File::getName).collect(Collectors.toSet());
+  }
+
+  private List<PatternLocation> convertToPatternLocations(String fileName) {
+    return convertToPatternLocations(Set.of(fileName));
+  }
+
+  private List<PatternLocation> convertToPatternLocations(Set<String> listOfFileNames) {
+    var specificationLoader = new SpecificationLoader(SpecificationLoader.DEFAULT_SPECIFICATION_LOCATION, listOfFileNames);
+
+    return specificationLoader.getRulesMappedToKey().entrySet().stream()
+      .map(this::toPatternLocation)
+      .flatMap(Collection::stream)
+      .collect(Collectors.toList());
+  }
+
+  private List<PatternLocation> toPatternLocation(Map.Entry<String, List<Rule>> entry) {
+    return entry.getValue().stream()
+      .map(this::ruleToRegexes)
+      .flatMap(Collection::stream)
+      .collect(Collectors.toList());
+  }
+
+  private List<PatternLocation> ruleToRegexes(Rule rule) {
+    String rspecKey = rule.getRspecKey();
+    var secretRuleId = rule.getId();
+
+    List<PatternLocation> patternLocations = new ArrayList<>();
+    if (rule.getDetection().getMatching() != null) {
+      patternLocations.add(new PatternLocation(rspecKey, secretRuleId, "matching", rule.getDetection().getMatching().getPattern()));
+    }
+
+    Match context = rule.getDetection().getMatching().getContext();
+    addMatchToRegexes(context, patternLocations, rspecKey, secretRuleId);
+
+    if (rule.getDetection().getPost() != null && rule.getDetection().getPost().getPatternNot() != null) {
+      rule.getDetection().getPost().getPatternNot().stream()
+        .map(pattern -> new PatternLocation(rspecKey, secretRuleId, "post", pattern))
+        .forEach(patternLocations::add);
+    }
+
+    // TODO this method needs to be extended when implementing SONARTEXT-71 Implement negation for auxiliary patterns
+    return patternLocations;
+  }
+
+  private void addMatchToRegexes(Match match, List<PatternLocation> regexes, String rspecKey, String secretRuleId) {
+    if (match instanceof AuxiliaryPattern) {
+      AuxiliaryPattern auxiliaryPattern = (AuxiliaryPattern) match;
+      var p = new PatternLocation(rspecKey, secretRuleId, "auxiliary", auxiliaryPattern.getPattern());
```

Comment:
`"auxiliary"` Could be changed to `"auxiliary-" + auxiliaryPattern.getType().toString()`